### DISCUSSION TITLE
Client vs server side counting + API items limit

### DISCUSSION THREAD

**Discussion by luciandavidescu** [umami-software/umami#1780] [Q&A] [answered]
This is simply brilliant!

Two quick questions:

1. To tell between users for bounce-rate calculation, does the client-side script use more info (beside screen size) then what's in the collect API headers + payload?
// to decide if server-side counting makes sense.

2. Is it possible to limit the number of returned items in the metrics API? 
// could be useful for websites with a lot of content.

**Comment by briancao** [comment, marked as answer]
1. The logic to identify users can be found in our session lib. It's current a mix of websiteUuid, hostname, ip, and userAgent.

https://github.com/umami-software/umami/blob/master/lib/session.js

2.  Are you talking about filters?  You can add filters by clicking on the individual metrics you want to drill into.  An example can be found on our demo site.

https://app.umami.is/share/8rmHaheU/umami.is?url=%2Fdocs%2Fgetting-started&referrer=google.com

**Reply by luciandavidescu** [reply]
1. Great (I sometimes even throw in user-language, to make a cid hash, but not sure really how much entropy that adds).
2. I mean the number of items in the json object returned from the API http request. If there were 1000, only the first 10 may be needed so I'm looking for a query arg such as '&limit=10'
